sensu/plugins/metric-os-api.py

```diff
@@ -0,0 +1,178 @@
+#
+# return CRITICAL if any commands fail
+# return WARNING if no commands fail, but one or more timeout
+#
+#
+# Jose L Coello Enriquez <jlcoello@us.ibm.com>
+
+from argparse import ArgumentParser
+import re
+import sys
+import time
+import subprocess
+import socket
+import os
+import json
+import requests
+
+STATE_OK = 0
+STATE_WARNING = 1
+STATE_CRITICAL = 2
+
+state = STATE_OK
+
+metric_data = ''
+
+DEFAULT_SCHEME = '{}'.format(socket.gethostname())
+
+
+def run(command):
+    p = subprocess.Popen(command, stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE)
+    stdout, stderr = p.communicate()
+    stdout = stdout.strip()
+    if p.returncode == 0 and stdout:
+        return stdout
+    else:
+        sys.exit(STATE_WARNING)
+
+def request_token(url, method='GET', retries=2, **kwargs):
+    r = None
+    try:
+        for i in range(retries + 1):
+            if i > 0:
+                time.sleep(2)
+            r = requests.request(method, url, **kwargs)
+            if r.status_code < 500:
+                break
+    except requests.exceptions.RequestException as e:
+        sys.exit(STATE_CRITICAL)
+
+    return r
+
+def request(scheme, url, metric, timestamp, method='GET', retries=2,**kwargs):
+    global state
+    global metric_data
+
+    start = time.time()
+    r = None
+    try:
+        for i in range(retries + 1):
+            if i > 0:
+                time.sleep(2)
+            r = requests.request(method, url, **kwargs)
+            if r.status_code < 500:
+                break
+    except requests.exceptions.RequestException as e:
+        print "{0} returned {1}".format(url, e)
+
+    if r is None:
```

Comment:
3 cases:
- if r is not passed anything so it was not able to connect after certain number of tries to call the api
- If a connection was possible and the http code returned was not 200, any fail case we change the state
-everything went fine 

So pretty much the first 2 checks are done to differentiate between failing on the retries and able to request the api but none 200 code was return